### PR TITLE
[ONNX] Align behavior of ONNX Frontend operator ReduceMean-11, 13, 18 with original framework

### PR DESCRIPTION
**Pull request by rghvsh**
### Details:
 - Align behavior of ONNX Frontend operator ReduceMean-11, 13, 18 with original framework

### Tickets:
 - Closes https://github.com/openvinotoolkit/openvino/issues/20556

### CODE REVIEW

**Comment by gkrivor** [comment]
Hi, I refactored reduce.cpp and reduce.hpp. I think it should be more simple to add a new functionality.

Please, rebase after merge.

https://github.com/openvinotoolkit/openvino/pull/23429

**Comment by gkrivor** [comment]
Hi @rghvsh ,

Please, take a look on https://github.com/openvinotoolkit/openvino/pull/23475 as a reference

**Comment by rghvsh** [comment]
Hi @gkrivor! I had a few questions:

do we need to add opset 11 to reduce.hpp
do we need to add opset 13 to reduce.cpp and reduce.hpp
in ops_bridge do I explicitly mention opset 11,13,18 or type this
`register_operator("ReduceMean", VersionRange{1, 17}, op::set_1::reduce_mean);`
register_operator("ReduceLogSum", VersionRange::since(18), op::set_1::reduce_mean);

Thank You

**Comment by rghvsh** [comment]
Hi @gkrivor @mitruska @onnx do you have any suggestions

**Comment by rghvsh** [comment]
Hi @gkrivor make changes today

**Comment by gkrivor** [comment]
@rghvsh as you can see, your change is very important and fix a lot of failing tests:
https://github.com/openvinotoolkit/openvino/actions/runs/8875797101/job/24366553495?pr=23148
Let's complete it asap. You need to remove failing cases from https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py

File: src/frontends/onnx/tests/models/ReduceMean.prototxt
Comment (mitruska) [inline src/frontends/onnx/tests/models/ReduceMean.prototxt:1]: The test model has been added, but there is no unit test code for it, so it's not tested.
For each added model there should be corresponding test in  [openvino/src/frontends/onnx/tests/onnx_import.in.cpp](https://github.com/openvinotoolkit/openvino/blob/59c54cb60e774a60b275913014517a051c0b1ec8/src/frontends/onnx/tests/onnx_import.in.cpp#L1018-L1019).
Also please keep the names of the onnx test models files aligned (snake_case).

Example:
https://github.com/openvinotoolkit/openvino/blob/59c54cb60e774a60b275913014517a051c0b1ec8/src/frontends/onnx/tests/onnx_import.in.cpp#L1018-L1019

**Comment by gkrivor** [comment]
@rghvsh could you add some tests and remove test which are working after your changes (take a look on my previous comment)?

**Comment by gkrivor** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Pull request merged**